internal/cluster/distribution/eks/node_pool_test.go

```diff
@@ -0,0 +1,116 @@
+// Copyright © 2020 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package eks
+
+import (
+	"testing"
+
+	"emperror.dev/errors"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/banzaicloud/pipeline/internal/cluster"
+)
+
+func TestNodePoolSizeValidation(t *testing.T) {
+	base := NewNodePool{
+		Name:         "pool",
+		InstanceType: "c5.large",
+		Image:        "ami",
+	}
+
+	t.Run("PoolZeroSize", func(t *testing.T) {
+		pool := base
+		pool.Size = 0
+		pool.Autoscaling.Enabled = false
+
+		err := pool.Validate()
+		assert.IsType(t, cluster.ValidationError{}, errors.Cause(err))
+	})
+
+	t.Run("PoolSizeNegative", func(t *testing.T) {
```

Comment:
**Recommended**: renaming this case **SHOULD** be considered for clarity purposes, the size is positive here, only the minimum size is negative.
```suggestion
	t.Run("PoolMinSizeNegative", func(t *testing.T) {
```